analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TooManyGenericParameters.DefaultValues.cs

```diff
@@ -77,7 +77,10 @@ namespace MyLib
 {
     public abstract class FrameworkBaseClass<T1, T2, T3> // Noncompliant
     {
-
+        void Method()
```

Comment:
Educational: can you help me understand why this test is in `TooManyGenericParameters.Default.Values.cs`?
I am a bit confused by the fact that this test takes place in a `abstract` class, in a `namespace MyLib`, but neither of those aspects seems relevant for the test on the generic lambda. Or am I missing something?

Reply:
`TooManyGenericParameters.DefaultValues.cs` is the base test script for this rule, it's differentiated between CustomValues and DefaultValues, the first one has a custom MaxNumberOfGenericParametersInClass parameter set to 4, as MaxNumberOfGenericParametersInMethod.
I added the test inside an abstract class for no particular reason, it was quite random TBH, the location was not important and I just wanted to make sure that the rule was raising for a generic lambda.